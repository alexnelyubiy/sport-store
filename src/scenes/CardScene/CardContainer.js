import {
  compose, setDisplayName, mapProps
} from 'recompose';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { getCartProducts, getPricesSum } from 'selectors';
import CardScene from './CardScene';

export const enhance = compose(
  setDisplayName('ProductsContainer'),
  connect(
    state => ({
      cartProducts: getCartProducts(state),
      pricesSum: getPricesSum(state)
    }), null
  ),

  mapProps(props => ({
    ...props,
    cartProducts: props.cartProducts.toJS()
  })),
);

export default enhance(CardScene);
