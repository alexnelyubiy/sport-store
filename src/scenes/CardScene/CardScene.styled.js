import styled from "styled-components";

export const TotalWrap = styled.div`
  padding-top: 5px;
  border-top: 2px solid #d8d8d8;
  width: 100%;
  text-align: right;
  color: grey;
  font-size: 20px;
`;
