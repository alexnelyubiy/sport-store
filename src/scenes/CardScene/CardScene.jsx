import React from "react";
import Master from "scenes/Master";
import CartProductContainer from "components/widgets/CartProduct";
import UserInfoContainer from "components/widgets/UserInfo";
import { ProductsWrapper } from "../ProductsScene/ProductsScene.styled";
import { TotalWrap } from "./CardScene.styled";

const displayName = "CardScene";

function CardScene({ cartProducts, isCart, pricesSum }) {
  return (
    <Master title="Cart">
      <ProductsWrapper>
        {cartProducts.map(product => (
          <CartProductContainer key={product.id} isCart={isCart} product={product} />
        ))}
      </ProductsWrapper>
      <TotalWrap>
        {cartProducts.length !== 0 ? <span>Total price: {pricesSum}$</span> : <span>Your cart is empty</span>}
      </TotalWrap>
      {
        cartProducts.length > 0 ? <UserInfoContainer /> : null
      }
    </Master>
  );
}

CardScene.displayName = displayName;

export default CardScene;
