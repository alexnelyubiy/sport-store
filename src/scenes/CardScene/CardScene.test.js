import React from 'react';
import { shallow } from 'enzyme';
import CardScene from './CardScene';

const testProps = {
  handleSubmit: jest.fn(),
  cartProducts: [{name: "Item 1", price: 290}, {name: "Item 2", price: 20}],
  isCart: true,
  pricesSum: 310
}
describe('Given the CardScene component', () => {
  describe('when the component is rendered', () => {
    let component;
  
    beforeEach(() => {
			component = shallow(<CardScene { ...testProps } />);
		});
  
    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
    describe('and cartProducts is empty', () => {
			beforeAll(() => {
				testProps.cartProducts = [];
			});

			it('should match the snapshot with emty cart', () => {
				expect(component).toMatchSnapshot();
			});
		});
  });
});
//

