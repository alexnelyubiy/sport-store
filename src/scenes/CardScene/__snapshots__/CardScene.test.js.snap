// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CardScene component when the component is rendered and cartProducts is empty should match the snapshot with emty cart 1`] = `
<Master
  title="Cart"
>
  <ProductsScenestyled__ProductsWrapper />
  <CardScenestyled__TotalWrap>
    <span>
      Your cart is empty
    </span>
  </CardScenestyled__TotalWrap>
</Master>
`;

exports[`Given the CardScene component when the component is rendered should render the component with provided props 1`] = `
<Master
  title="Cart"
>
  <ProductsScenestyled__ProductsWrapper>
    <CartProductContainer
      isCart={true}
      product={
        Object {
          "name": "Item 1",
          "price": 290,
        }
      }
    />
    <CartProductContainer
      isCart={true}
      product={
        Object {
          "name": "Item 2",
          "price": 20,
        }
      }
    />
  </ProductsScenestyled__ProductsWrapper>
  <CardScenestyled__TotalWrap>
    <span>
      Total price: 
      310
      $
    </span>
  </CardScenestyled__TotalWrap>
  <ProductContainer />
</Master>
`;
